thegame/engine/engine.py

```diff
@@ -72,33 +90,102 @@ def _main_loop(self):
             # Handle events if any have come in.
             if number_of_events > 0:
 
-                # Each event should be independent of the other,
-                # thus we can process each one as if the others
-                # didn't happen.
-                event_pool = ThreadPool(processes=number_of_events)
-
-                exception_check = []
-                for event in events:
-                    if self.running:
-                        exception_check.append(
-                            (
-                                event_pool.apply_async(
-                                    self._handle_event, args=(event,)
-                                ),
-                                event,
-                            )
-                        )
+                self._schedule_events(events=events, number_of_events=number_of_events)
 
-                for possible_exception, event in exception_check:
-                    try:
-                        possible_exception.get()
-                    except Exception as e:
-                        self.running = False
-                        logging.exception(
-                            f"Caught exception while handling "
-                            f"'{pygame.event.event_name(event.type)}' event: '{e}'."
-                        )
-                        break
+            # TODO: Find a more efficient method of this.
+            #   clearing then re-adding all the sprites from
+            #   the group seems inefficient.
+            game_sprites.empty()
+
+            # Discover which portion of the screen needs to be drawn
+            if self.context.active_menu is not None:
+                menu_sprite = self.context.active_menu.menu_image
+                game_sprites.add(menu_sprite)
+            else:
+                onscreen_sprites = []
+                fov_tile_sheet = self.context.camera.get_camera_fov(
+                    self.context.active_screen
+                )
+                logging.debug(
+                    f"Got the following fov_tile_sheet: {pprint.pformat(fov_tile_sheet, indent=4)}"
+                )
+
+                # Currently, this is done with (4 * camera_width * camera_height) loops.
+                # the only way to reduce this is to not loop over rows full of Nones, but
+                # this check isn't worth the gain.
+                # Updates each sprites' x and y position.
+                for layer in fov_tile_sheet:
```

Comment:
Should this be done in reverse order?
Right now sprites in the bottom layer will be drawn after sprites in an upper layer, which is the opposite of the desired functionality.

Reply:
More than likely, will test